src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/gui/ProxyControlGui.java

```diff
@@ -691,6 +698,17 @@ void enableRestart() {
         }
     }
 
+    void setSetCounters() {
+        if ((counterValue.getText().length() > 0) && NumberUtils.isParsable(counterValue.getText())) {
+            Proxy.setCounter(Integer.parseInt(counterValue.getText()));
+        } else {
+            JOptionPane.showMessageDialog(this,
+                    //"Counter isn't a number value.", "Error on counter value", JOptionPane.WARNING_MESSAGE);
+                    JMeterUtils.getResString("proxy_settings_counter_error_digits"), // $NON-NLS-1$
+                    JMeterUtils.getResString("proxy_settings_counter_error_invalid_data"), // $NON-NLS-1$
+                    JOptionPane.WARNING_MESSAGE);
```

Comment:
Could you add the problematic value to the error message?

Reply:
Will it support `${...}` expressions?

Reply:
@vlsi what is ${...} expressions? I don't understand
Not sure that add the problematic value is needed, specially if the value is empty. IMO only display a warning is ok.

Reply:
E.g. something like `${__P(staringValue)}`

Reply:
Ok, I see the number was already parsed without evaluating the properties, so supporting `${..}` there would be a different issue

Reply:
Seems usual to not evaluate the value against the JMeter function for the Proxy settings. Proxy is use only on GUI mode, not in load tests, so it's not need to evaluate the value if ${...} expression is present. Are you ok with this?

Reply:
I have no use case for evaluation `${..}` there yet, however, I think having uniform support for `${..}` would be nice

Reply:
This PR aims first to correct the NFE (which is a technical error), and not to make improvements in the Counter field. If there is a need to add functionality for evaluating JMeter functions, that will be the subject of another ticket. Can I have your LGTM?

Reply:
That works for me. However, would you please include the problematic value itself in the error message?